4.1.0 Release

stanfordquan opened the pull request:
notes to follow

stanfordquan commented on gslab_scons/misc.py:
```diff
@@ -11,39 +11,42 @@
 
-def scons_debrief(target, env):
-    '''Execute functions after SCons has built all targets'''
+def scons_debrief(args):
```
This introduce backward incompatibility, since the user has to modify SConstruct. @Shun-Yang I don't have a great grasp on this function and you were the last one to touch it, can you add a quick fix?

stanfordquan replied:
^ Comment as part of #130 #133

Shun-Yang replied:
We can change `args['log']` back into the function argument as `log = 'state_of_repo.log'` but the reason we moved it is that this looks more compact and natural.

stanfordquan replied:
We had a discussion in the lab just now. Our solution is to provide a user-advisory that you cannot pass `env` and `log` to the function any longer and have to pass `args`, at which point we point to the current SConstruct in `template`. So we're fine with both 1) backward incompatibility and 2) coupled gslab_python/template for this particular problem.

stanfordquan commented:
# gslab_python 4.1.0 Release Notes

## Features

- We added a LaTeX builder and an "Anything"-builder. (#96 #97)

- We added code to reconfigure the SCons cache (#99)

- We add a target-checker after a builder has run (#95)

- `gslab_scons` features now use a local version of SCons by default (#111 #126)

- We add a feature to produce `state_of_inputs` (#131) 

- `gslab_python` installation now delete old local versions (#122)

## Misc

- We move top-level logs to `release/.` or a user-specified location (#93)

- We allow suffixes to the name of builder logs (#94)

- We updated R syntax (#98) and Stata-executable searching behavior (#92)

arosenbe commented:
@stanfordquan, I think this is ready to go. Do you want to make the pull?
